geoh5py/data/data_type.py

```diff
@@ -153,43 +132,57 @@ def color_map(self, color_map: ColorMap | dict | np.ndarray | None):
         self.workspace.update_attribute(self, "color_map")
 
     @classmethod
-    def for_x_data(cls, workspace: Workspace) -> DataType:
+    def find_or_create(
+        cls,
+        workspace: Workspace,
+        uid: UUID | None = None,
+        entity_class: type | None = None,
+        **kwargs,
+    ):
         """
-        Get the data type for x data.
+        Find or creates an EntityType with given uid that matches the given
+        Group implementation class.
 
-        :param workspace: An active Workspace.
+        It is expected to have a single instance of EntityType in the Workspace
+        for each concrete Entity class.
 
-        :return: A new instance of DataType.
-        """
-        return cls._for_geometric_data(
-            workspace, GeometricDataConstants.x_datatype_uid()
-        )
+        To find an object, the kwargs must contain an existing 'uid' keyword,
+        or a 'entity_class' keyword, containing an object class.
 
-    @classmethod
-    def for_y_data(cls, workspace: Workspace) -> DataType:
+        :param workspace: An active Workspace class
+        :param uid: The unique identifier of the entity type.
+        :param entity_class: The class of the entity.
+        :param kwargs: The attributes of the entity type.
+
+        :return: EntityType
         """
-        Get the data type for y data.
+        primitive_type = kwargs.get("primitive_type", None)
+        if isinstance(primitive_type, str) and primitive_type.upper() == "GEOMETRIC":
```

Comment:
general question: why geometric type should be considered as a different datatype?

Reply:
It's the only data type that has a fixed uuid